### PR TITLE
DEV: Convert poll modals to new component-based API

### PR DESCRIPTION
**Pull request by davidtaylorhq**
(Requires https://github.com/discourse/discourse/pull/21304)

### CODE REVIEW

**Review by CvX — APPROVED** [review]

File: plugins/poll/assets/javascripts/discourse/components/modal/poll-ui-builder.hbs [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+                  @icon="trash-alt"
+                  @action={{action "removeOption" option}}
+                />
+              {{/if}}
+            </div>
+          {{/each}}
+
+          <div class="poll-option-controls">
+            <DButton
+              @class="btn-default poll-option-add"
```
Comment (janzenisaac) [inline plugins/poll/assets/javascripts/discourse/components/modal/poll-ui-builder.hbs:81]:
```suggestion
              class="btn-default poll-option-add"
```

nit

File: plugins/poll/assets/javascripts/discourse/components/modal/poll-ui-builder.hbs [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+          </label>
+        </div>
+      {{/unless}}
+    {{/if}}
+  </:body>
+  <:footer>
+    <DButton
+      @action={{action "insertPoll"}}
+      @icon="chart-bar"
+      @class="btn-primary insert-poll"
```
Comment (janzenisaac) [inline plugins/poll/assets/javascripts/discourse/components/modal/poll-ui-builder.hbs:232]:
```suggestion
      class="btn-primary insert-poll"
```

File: plugins/poll/assets/javascripts/discourse/components/modal/poll-ui-builder.hbs [hunk 3]
```diff
@@ -0,0 +1,251 @@
...
+      @label="poll.ui_builder.insert"
+      @disabled={{this.disableInsert}}
+    />
+
+    <DButton @label="cancel" @class="btn-flat" @action={{@closeModal}} />
+
+    <DButton
+      @action={{action "toggleAdvanced"}}
+      @class="btn-default show-advanced"
```
Comment (janzenisaac) [inline plugins/poll/assets/javascripts/discourse/components/modal/poll-ui-builder.hbs:241]:
```suggestion
      class="btn-default show-advanced"
```

File: plugins/poll/assets/javascripts/discourse/components/modal/poll-ui-builder.js [hunk 4]
```diff
@@ -19,46 +18,26 @@ const VOTE_POLL_RESULT = "on_vote";
...
-export default class PollUiBuilderController extends Controller.extend(
-  ModalFunctionality
-) {
+export default class PollUiBuilderController extends Component {
   showAdvanced = false;
   pollType = REGULAR_POLL_TYPE;
-  pollTitle = "";
-  pollOptions = null;
-  pollOptionsText = null;
+  pollTitle = null;
```
Comment (janzenisaac) [inline plugins/poll/assets/javascripts/discourse/components/modal/poll-ui-builder.js:24]: Seeing this is `undefined` by default, can we just do"

```suggestion
  pollTitle;
```

File: plugins/poll/assets/javascripts/discourse/components/modal/poll-ui-builder.js [hunk 5]
```diff
@@ -19,46 +18,26 @@ const VOTE_POLL_RESULT = "on_vote";
...
+  pollOptions = [EmberObject.create({ value: "" })];
+  pollOptionsText = "";
   pollMin = 1;
   pollMax = 2;
   pollStep = 1;
   pollGroups = null;
   pollAutoClose = null;
```
Comment (janzenisaac) [inline plugins/poll/assets/javascripts/discourse/components/modal/poll-ui-builder.js:31]: Dito

**Review by janzenisaac — APPROVED** [review]

**Pull request merged**